server-spi-private/src/main/java/org/keycloak/services/clientpolicy/ClientPoliciesUtil.java

```diff
@@ -0,0 +1,747 @@
+        }
+        if (realm == null) {
+            throw new ClientPolicyException("realm not specified.", "NA");
+        }
+
+        // deserialize existing profiles (json -> representation)
+        ClientProfilesRepresentation existingProfilesRep = null;
+        String existingProfilesJson = session.clientPolicy().getClientProfilesJsonString(realm);
+        if (existingProfilesJson != null) {
+            try {
+                existingProfilesRep = convertClientProfilesJsonToRepresentation(existingProfilesJson);
+            } catch (ClientPolicyException e) {
+                throw new ClientPolicyException(e.getError(), e.getErrorDetail());
+            }
+            if (existingProfilesRep == null) {
+                existingProfilesRep = new ClientProfilesRepresentation();
+            }
+        } else {
+            existingProfilesRep = new ClientProfilesRepresentation();
+        }
+
+        // no profile contained (it is valid)
+        // back to initial builtin profiles
+        List<ClientProfileRepresentation> proposedProfileRepList = proposedProfilesRep.getProfiles();
+        if (proposedProfileRepList == null || proposedProfileRepList.isEmpty()) {
+            proposedProfileRepList = new ArrayList<>();
+            proposedProfilesRep.setProfiles(new ArrayList<>());
+        }
+
+        // duplicated profile name is not allowed.
+        if (proposedProfileRepList.size() != proposedProfileRepList.stream().map(i->i.getName()).distinct().count()) {
+            throw new ClientPolicyException("proposed client profile name duplicated.", "NA");
+        }
+
+        // construct updating profiles from existing profiles and proposed profiles
+        ClientProfilesRepresentation updatingProfilesRep = new ClientProfilesRepresentation();
+        updatingProfilesRep.setProfiles(new ArrayList<>());
+        List<ClientProfileRepresentation> updatingProfileList = updatingProfilesRep.getProfiles();
+
+        // add existing builtin profiles to updating profiles
+        List<ClientProfileRepresentation> existingProfileList = existingProfilesRep.getProfiles();
+        if (existingProfileList != null && !existingProfileList.isEmpty()) {
+            existingProfileList.stream().filter(i->i.isBuiltin()).forEach(i->updatingProfileList.add(i));
+        }
+
+        for (ClientProfileRepresentation proposedProfileRep : proposedProfilesRep.getProfiles()) {
+            if (proposedProfileRep.getName() == null) {
+                continue;
+            }
+
+            // ignore proposed builtin profile because builtin profile cannot added/deleted/modified.
+            if (proposedProfileRep.isBuiltin() != null && proposedProfileRep.isBuiltin()) {
+                logger.warnv("{0} builtin profile included in proposed profiles. Just ignore it. profile name = {1}", LOGMSG_PREFIX, proposedProfileRep.getName());
+                continue;
+            }
+
+            // basically, proposed profile totally overrides existing profile
```

Comment:
This comment is confusing as it doesn't work in a way specified in this comment. Currently it works in a way that builtin profiles are always added to the realm and it's not possible to remove or disable them. However at this moment, I am able to add the proposed profile called `builtin-default-profile`, which is probably not correct due the fact that is is same as the name of the existing builtin profile.

I suggest that things work in a way that:
- Builtin profiles are always added (exactly as you did in this PR)
- In case that `proposedProfiles` contains some profile with same name as the builting profile, the error will be thrown. So if administrator sends the profile with a name `builtin-default-profile`, the error is thrown.

So this comment can likely change as well (or be deleted? Whatever you prefer)
WDYT?

Reply:
@mposolda Yes, what you suggested are the one I had wanted to do. The current codes allows the builtin profile to be overridden by the proposed one whose name is the same as in the builtin profile. This is the problem.

Therefore, I would like to fix the codes as you proposed, namely checking whether the propose profiles name is the same as in the built profile and throws the error if so. The current comment remains intact.

Reply:
Thanks!